osutil/disks/disks_linux_test.go

```diff
@@ -891,3 +891,74 @@ func (s *diskSuite) TestDiskFromMountPointDecryptedDevicePartitionsHappy(c *C) {
 	c.Assert(err, IsNil)
 	c.Assert(matches, Equals, true)
 }
+
+func (s *diskSuite) TestRootMountPointsForPartition(c *C) {
+	const (
+		validRootMnt1    = "130 30 42:1 / /run/mnt/ubuntu-data rw,relatime shared:54 - ext4 /dev/vda3 rw\n"
+		validRootMnt2    = "130 30 42:1 / /run/mnt/foo-other-place rw,relatime shared:54 - ext4 /dev/vda3 rw\n"
+		validNonRootMnt1 = "130 30 42:1 /subdir /run/mnt/other-ubuntu-data rw,relatime shared:54 - ext4 /dev/vda3 rw\n"
+		validNonRootMnt2 = "130 30 42:1 /subdir2 /run/mnt/other-ubuntu-data-other-other rw,relatime shared:54 - ext4 /dev/vda3 rw\n"
+	)
+
+	tt := []struct {
+		maj, min  int
+		mountinfo string
+		exp       []string
+		comment   string
+	}{
+		{
+			comment:   "single valid root mountpoint",
+			mountinfo: validRootMnt1,
+			exp:       []string{"/run/mnt/ubuntu-data"},
+		},
+		{
+			comment:   "multiple valid root mountpoints",
```

Comment:
I feel a bit dumb asking - but under which scenarios do we have multiple valid root mount points?

Reply:
no worries at all, this happens when we use bind mounts of another root mount for example such as these four mounts we have for a uc20 VM:
```
ubuntu@ubuntu:~$ cat /proc/self/mountinfo | grep 252:2
128 31 252:2 / /run/mnt/ubuntu-seed rw,relatime shared:54 - vfat /dev/vda2 rw,fmask=0022,dmask=0022,codepage=437,iocharset=iso8859-1,shortname=mixed,errors=remount-ro
57 141 252:2 / /boot/efi rw,relatime shared:54 - vfat /dev/vda2 rw,fmask=0022,dmask=0022,codepage=437,iocharset=iso8859-1,shortname=mixed,errors=remount-ro
353 75 252:2 / /var/lib/snapd/seed ro,relatime shared:54 - vfat /dev/vda2 rw,fmask=0022,dmask=0022,codepage=437,iocharset=iso8859-1,shortname=mixed,errors=remount-ro
354 152 252:2 / /writable/system-data/var/lib/snapd/seed rw,relatime shared:54 - vfat /dev/vda2 rw,fmask=0022,dmask=0022,codepage=437,iocharset=iso8859-1,shortname=mixed,errors=remount-ro
ubuntu@ubuntu:~$ ls -lah /boot/efi/
total 14K
drwxr-xr-x 6 root root  512 Dec 31  1969 .
drwxr-xr-x 6 root root   70 Sep 28 06:55 ..
drwxr-xr-x 4 root root  512 Sep 30 10:44 EFI
-rwxr-xr-x 1 root root  11K Sep 30 15:45 NvVars
drwxr-xr-x 3 root root  512 Sep 30 10:44 data
drwxr-xr-x 2 root root 1.0K Sep 30 10:44 snaps
drwxr-xr-x 3 root root  512 Sep 30 10:44 systems
ubuntu@ubuntu:~$ ls -lah /run/mnt/ubuntu-seed/
total 14K
drwxr-xr-x  6 root root  512 Dec 31  1969 .
drwxr-xr-x 10 root root  200 Sep 30 15:45 ..
drwxr-xr-x  4 root root  512 Sep 30 10:44 EFI
-rwxr-xr-x  1 root root  11K Sep 30 15:45 NvVars
drwxr-xr-x  3 root root  512 Sep 30 10:44 data
drwxr-xr-x  2 root root 1.0K Sep 30 10:44 snaps
drwxr-xr-x  3 root root  512 Sep 30 10:44 systems
ubuntu@ubuntu:~$ ls -lah /var/lib/snapd/seed
total 18K
drwxr-xr-x  6 root root  512 Dec 31  1969 .
drwxr-xr-x 15 root root 4.0K Sep 30 15:46 ..
drwxr-xr-x  4 root root  512 Sep 30 10:44 EFI
-rwxr-xr-x  1 root root  11K Sep 30 15:45 NvVars
drwxr-xr-x  3 root root  512 Sep 30 10:44 data
drwxr-xr-x  2 root root 1.0K Sep 30 10:44 snaps
drwxr-xr-x  3 root root  512 Sep 30 10:44 systems
ubuntu@ubuntu:~$ ls -lah /writable/system-data/var/lib/snapd/seed/
total 18K
drwxr-xr-x  6 root root  512 Dec 31  1969 .
drwxr-xr-x 15 root root 4.0K Sep 30 15:46 ..
drwxr-xr-x  4 root root  512 Sep 30 10:44 EFI
-rwxr-xr-x  1 root root  11K Sep 30 15:45 NvVars
drwxr-xr-x  3 root root  512 Sep 30 10:44 data
drwxr-xr-x  2 root root 1.0K Sep 30 10:44 snaps
drwxr-xr-x  3 root root  512 Sep 30 10:44 systems
```

All the mounts are for the /dev/vda2 (ubuntu-seed) partition's root directory